contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/zonal/MostCentralDrtZoneTargetLinkSelector.java

```diff
@@ -0,0 +1,51 @@
+/* *********************************************************************** *
+ * project: org.matsim.*
+ * Controler.java
+ *                                                                         *
+ * *********************************************************************** *
+ *                                                                         *
+ * copyright       : (C) 2007 by the members listed in the COPYING,        *
+ *                   LICENSE and WARRANTY file.                            *
+ * email           : info at matsim dot org                                *
+ *                                                                         *
+ * *********************************************************************** *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *   See also COPYING, LICENSE and WARRANTY file                           *
+ *                                                                         *
+ * *********************************************************************** */
+
+package org.matsim.contrib.drt.analysis.zonal;
+
+import org.matsim.api.core.v01.network.Link;
+import org.matsim.core.network.NetworkUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class MostCentralDrtZoneTargetLinkSelector implements DrtZoneTargetLinkSelector{
+
+	Map<DrtZone,Link> targetLinks = new HashMap<>();
+
+	@Override
+	public Link selectTargetLinkFor(DrtZone zone) {
+		if(this.targetLinks.containsKey(zone)) return this.targetLinks.get(zone);
```

Comment:
since we do not have so many zones, it's simpler to fill the map eagerly (and less error-prone).

Reply:
I put the map filling into the constructor - hope this is what you meant.